src/services/helpers/types.ts

```diff
@@ -22,11 +22,26 @@ export interface AlertItem {
 
 export interface FactoryResolver {
   v: string;
-  source: string;
+  source?: string;
   devfile: api.che.workspace.devfile.Devfile;
   location?: string;
+  'scm_info'?: FactoryResolverScmInfo;
 }
 
+export type FactoryResolverScmInfo = {
+  'clone_url': string;
+  'scm_provider': string;
+  'branch'?: string;
+};
+
+export type ProjectV2Source = {
```

Comment:
```suggestion
export type DevfileV2ProjectSource = {
```
^ may be a bit clearer but feel free to ignore it, anyway we would need to replace it with type from devfile/api

Reply:
I have fixed it.